scripts/dc2/create_lensed_host_ic.py

```diff
@@ -0,0 +1,176 @@
+import os
+import numpy as np
+import pandas as pd
+import argparse
+from sqlalchemy import create_engine
+from lsst.sims.catUtils.utils import ObservationMetaDataGenerator
+from desc.sims.GCRCatSimInterface import get_obs_md
+from lsst.sims.utils import angularSeparation
+from dc2_utils import instCatUtils
+
+__all__ = ['hostImage']
+
+class hostImage(instCatUtils):
+    """Takes FITS stamps and includes them in instance catalogs
+    hostImage takes the following arguments:
+    ra_center: Right ascension of the center of the field (in degrees)
+    dec_center: Declination of the center of the field (in degrees)
+    fov: Field-of-view angular radius (in degrees).  2 degrees will cover
+        the LSST focal plane."""
+
+    def __init__(self, obs_md, fov):
+
+        self.ra = obs_md.pointingRA
+        self.dec = obs_md.pointingDec
+        self.radius = fov
+        self.bandpass = obs_md.bandpass
+        self.obs_md = obs_md
+
+        self.bandpass_lookup = {'u': 0, 'g': 1, 'r': 2, 'i': 3, 'z': 4, 'y': 5}
+
+
+    def format_catalog(self, df_line, fits_file_name, image_dir):
+        """
+        Formats the output instance catalog to include entries for the FITS 
+        stamps produced by generate_lensed_host_***.py
+		
+		Parameters: 
+		-----------
+		df_line: string
+			pandas data frame line
+		fits_file_name: string
+			the filename of the FITS stamp
+		image_dir: string 
+			the location of the FITS stamps 
+		Returns:
+		-----------
+		cat_str: a string containing the line of parameters to go into an instance file
+        """
+
+        fits_info = fits_file_name.split('_')
+        lens_id = np.int(fits_info[0])
+        sys_magNorm_list = np.array(fits_info[1:-1], dtype=np.float)
+        sys_magNorm = sys_magNorm_list[self.bandpass_lookup[self.bandpass]]
+        gal_type = fits_info[-1].split('.')[0]
+
+        # galaxy_id = np.right_shift(lens_id, 10)
+        # galaxy_id *= 10000
+        # galaxy_id += 4*df_line['twinkles_system']
+        # sys_id = np.left_shift(galaxy_id, 10)
+        # if gal_type == 'bulge':
+        #     sys_id += 97
+        # elif gal_type == 'disk':
+        #     sys_id += 107
+        # else:
+        #     raise ValueError('Invalid Galaxy Component Type in filename')
+        sys_id = lens_id
+
+        cat_str = 'object %i %f %f %f %s %f 0 0 0 0 0 %s 0.01 0 CCM %f %f CCM %f %f\n' % (sys_id,
+                                                                      df_line['ra_lens'],
+                                                                      df_line['dec_lens'],
+                                                                      sys_magNorm,
+                                                                      df_line['sed_%s_host' % gal_type].decode('utf-8'),
+                                                                      df_line['redshift'],
+                                                                      str(image_dir)+'/'+fits_file_name,
```

Comment:
Would it be possible to write the part of the path that's relative to the `fits_stamp_dir` instead of the full path, e.g., writing just something like `agn_lensed_disks/1687651970800016_23.18015099726582_23.16421659284453_23.177045742542568_23.2013
49235772643_23.239502697720983_23.25863079851184_disk.fits`?  This way we can relocate the folder with all of the stamps.  imSim allows one to specify the folders to search for these FITS files at runtime.

Reply:
Ok, I shortened the paths.